pkg/scheduler/framework/plugins/noderesources/resource_allocation.go

```diff
@@ -72,28 +74,29 @@ func (r *resourceAllocationScorer) score(
 	return score, nil
 }
 
-// calculateResourceAllocatableRequest returns resources Allocatable and Requested values
-func calculateResourceAllocatableRequest(nodeInfo *framework.NodeInfo, pod *v1.Pod, resource v1.ResourceName, enablePodOverhead bool) (int64, int64) {
+// calculateResourceAllocatableRequest returns 3 parameters:
+// - 1st param: aggregated allocatable resource values on the node.
+// - 2nd param: aggregated requested resource values on the node.
+// - 3rd param: true if resource is scalar and pod request is zero; false otherwise.
+func calculateResourceAllocatableRequest(nodeInfo *framework.NodeInfo, pod *v1.Pod, resource v1.ResourceName, enablePodOverhead bool) (int64, int64, bool) {
 	podRequest := calculatePodResourceRequest(pod, resource, enablePodOverhead)
+	zeroScalarReq := schedutil.IsScalarResourceName(resource) && podRequest == 0
```

Comment:
what about just returning 0, 0 in this case? Then we don't need the boolean.

Reply:
I didn't follow - returning 0, 0 cannot tell whether it's a 0-requested resource and if it's an extended resource or not.

Reply:
The idea would be that a valid node (node that passed filters) shouldn't have allocatable 0 for a resource, unless the pod didn't request it.

So, returning 0, 0 is enough. Then, in line 59 you can just do `if sumAllocatable != 0`.
But this is a nit.

Also, the name `sumAllocatable` is not accurate, as it is not a sum at all.

Reply:
ok, got what you mean - even it's a node with extended res, if the pod doesn't request it, return the node's allocatable as 0.

This works, but set an implication here - using 0 allocatable to represent the semantic of "zero requested scalar resource". Let me add a comment here.

> Also, the name sumAllocatable is not accurate, as it is not a sum at all.

True, update the naming and comments.